src/textual/widgets/_markdown.py

```diff
@@ -500,19 +501,35 @@ class MarkdownFence(MarkdownBlock):
     def __init__(self, markdown: Markdown, code: str, lexer: str) -> None:
         self.code = code
         self.lexer = lexer
+        self.theme = "solarized-dark" if self.app.dark else "solarized-light"
         super().__init__(markdown)
 
-    def compose(self) -> ComposeResult:
-        from rich.syntax import Syntax
+    def _retheme(self) -> None:
+        """Swap between a dark and light theme when the mode changes."""
+        self.theme = "solarized-dark" if self.app.dark else "solarized-light"
+        code_block = self.query_one(Static)
+        code_block.renderable = Syntax(
```

Comment:
I feel that doubling up on the code that crates the `Syntax` object it something we should avoid.